### PR TITLE
Allow live content to be displayed on test / stage environments

### LINKED ISSUE #4822: Control the source of content data between Live/Test for TEST MAP pages
**Is your feature request related to a problem? Please describe.**
We want to be able to control whether we use PROD or TEST content data for MAP pages on the TEST environment. This is going to require additional mechanism being added to Mozart.

**Describe the solution you'd like**
Simorgh will accept a URL param to specify the source environment of content data (e.g. TEST or LIVE) and will pass this to Mozart. This will allow us to load TEST content or LIVE content based on this control parameter.

`renderer_env=int|test|live` e.g. http://www.example.com/pidgin/12345?renderer_env=live

NOTE Live Simorgh should never be able to load TEST content.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves #4822

May be required in conjunction with #4875 - click for further information

**Overall change:** 
It should be possible to view live content in test or staging environments, by appending a query string to the URL

**Code changes:**

- Pass urlPath (which does not contain the query string) instead of url (which is the entire string), so that react router continues to match the path regex.
- Adding tests

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

**Comment by karinathomasbbc** [comment]
I am not sure whether this supercedes #4875, or whether both are required.  This fix on its own works locally, and I would like to verify whether it also works in test/stage. If not, then the changes in #4875 will also be required

File: src/server/index.jsx [hunk 1]
```diff
@@ -209,7 +209,7 @@ server
   )
   .get('/*', cspInjectFun, async ({ url, headers, path: urlPath }, res) => {
     try {
-      const { service, isAmp, route, variant } = getRouteProps(routes, url);
+      const { service, isAmp, route, variant } = getRouteProps(routes, urlPath);
```
Comment (jroebu14) [inline src/server/index.jsx:214]: I think we need to carry out some careful checks inside of `getRouteProps` as it seems to do a lot and call a few other functions with regexy stuff. I'll take a look into this later today and let you know what I find out 🙂

**Review by ryanmccombe — APPROVED** [review]
👍

File: src/server/index.jsx (hunk 1 above)
Comment (chris-hinds) [reply src/server/index.jsx:214]: what is the difference between `url` and `path` here?

Comment (ryanmccombe) [reply src/server/index.jsx:214]: `req.url` includes querystring - `req.path` excludes it :)

**Review by jamesdonoh — COMMENTED** [review]
Is there anything that prevents the query from being passed in the live environment?

**Comment by karinathomasbbc** [comment]
> Is there anything that prevents the query from being passed in the live environment?

In this solution, passing urlPath instead of url means that the query params are stripped out (but only from the part of the code which checks that the path is a valid URL based on the different regexes we have for different page types).  This is an experimental solution - I want to see what happens in the test environment.  If it doesn't quite work then I might need to apply #4875 too.

**Review by jamesdonoh — APPROVED** [review]
Looks good, let's see what additional information we can get from the logs with this in place.

**Pull request merged**